examples/dfp_axis/src/main/java/dfp/axis/v201711/creativeservice/CreateCreativesFromTemplates.java

```diff
@@ -0,0 +1,215 @@
+import com.google.api.ads.dfp.lib.client.DfpSession;
+import com.google.api.ads.dfp.lib.utils.examples.ArgumentNames;
+import com.google.api.client.auth.oauth2.Credential;
+import java.io.IOException;
+import java.rmi.RemoteException;
+import java.util.Random;
+
+/**
+ * This example creates a new template creative for a given advertiser. To
+ * determine which creatives already exist, run GetAllCreatives.java. To
+ * determine which creative templates exist, run GetAllCreativeTemplates.java.
+ *
+ * Credentials and properties in {@code fromFile()} are pulled from the
+ * "ads.properties" file. See README for more info.
+ */
+public class CreateCreativesFromTemplates {
+
+  private static class CreateCreativesFromTemplatesParams extends CodeSampleParams {
+    @Parameter(names = ArgumentNames.ADVERTISER_ID, required = true,
+        description = "The ID of the advertiser (company) that all creatives will be assigned"
+            + " to.")
+    private Long advertiserId;
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param dfpServices the services factory.
+   * @param session the session.
+   * @param advertiserId the ID of the advertiser (company) that all creatives will be assigned to.
+   * @throws ApiException if the API request failed with one or more service errors.
+   * @throws RemoteException if the API request failed due to other errors.
+   * @throws IOException if unable to get media data from the URL.
+   */
+  public static void runExample(DfpServices dfpServices, DfpSession session, long advertiserId)
+      throws IOException {
+    // Get the CreativeService.
+    CreativeServiceInterface creativeService =
+        dfpServices.get(session, CreativeServiceInterface.class);
+
+    // Create creative size.
+    Size size = new Size();
+    size.setWidth(600);
+    size.setHeight(315);
+    size.setIsAspectRatio(false);
+
+    // Use the image banner with optional third party tracking template.
+    // To determine what other creative templates exist,
+    // run GetAllCreativeTemplates.java.
+    long creativeTemplateId = 10000680L;
+
+    // Create a template creative.
+    TemplateCreative templateCreative = new TemplateCreative();
+    templateCreative.setName("Template creative #" + new Random().nextInt(Integer.MAX_VALUE));
+    templateCreative.setAdvertiserId(advertiserId);
+    templateCreative.setCreativeTemplateId(creativeTemplateId);
+    templateCreative.setSize(size);
+
+    // Create the asset variable value.
+    AssetCreativeTemplateVariableValue assetVariableValue =
+        new AssetCreativeTemplateVariableValue();
+    assetVariableValue.setUniqueName("Imagefile");
+    
+    CreativeAsset asset = new CreativeAsset();
+    asset.setAssetByteArray(Media.getMediaDataFromUrl("https://goo.gl/3b9Wfh"));
+    // Filenames must be unique.
+    asset.setFileName(
+        String.format("image%s.jpg", new Random().nextInt(Integer.MAX_VALUE)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;nextInt&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 107. Consider formatting this data with &#39;%d&#39; instead.